### ISSUE TITLE
[ts-command-line] Support space-separated array values in CommandLineStringListParameter 

### ISSUE THREAD

**Issue by elliot-nelson** [microsoft/rushstack#2741]
## Summary

Today you can pass multiple values to a String List as follows:

```console
mycli --color red --color green
```

Let's also allow space-separated traditional positional args after a String List flag ("shortcut form"):

```console
mycli --color red green
```

## Details

The ability to support multiple positional args for a list parameter unlocks the ability to write node programs that accept shell-expanded wildcard patterns, for example:

```console
# bash/zsh automatically expands to: --files 1.log 2.log 3.log
./report-logs.js --files *.log

# desired `.values` = ['1.log', '2.log', '3.log']
```

In order to support this, a `CommandLineStringListParameter` should consume args until all args are consumed _or_ any arg starts with the character `'-'`.  This design choice does not allow passing a string value that starts with `-`, but it retains our ability to report a misspelled flag, which is much more likely.  (Users can still pass string values starting with dashes by using an equal form, i.e., `--path=-snakes-`, they just can't use the shortcut form.)

This new behavior should be supported by any `ListParameter` (for example, the proposed `CommandLineChoiceListParameter` in https://github.com/microsoft/rushstack/issues/2330).

Because we don't control the order a user uses to specify command-line options, this new behavior is incompatible with other parameters that attempt to "slurp" remaining args, like `CommandLineRemainder`.  (A solution is described in https://github.com/microsoft/rushstack/issues/2499; I believe `CommandLineRemainder` should be changed to follow the Unix tradition of slurping all args after the token `--` instead of just looking for unknown positional args.)  Once implemented, this would be supported in Rush as well, example:

```
# Run "custom-linter" bulk command
rush custom-linter --to @acme/Car @acme/Boat -- --fix --aggressive-newline

# toProject.values = ['@acme/Car', '@acme/Boat']
# remainder.values = ['--fix', '--aggressive-newline']
```

## Standard questions

| Question | Answer |
| -------- | -------- |
| Package name: | `@rushstack/ts-command-line` |
| Package version? | N/A |
| Operating system? | N/A |
| Would you consider contributing a PR? | Yes |
| Node.js version (`node -v`)? | 12.17.0 |

**Comment by octogonz** [comment]
>  (Users can still pass string values starting with dashes by using an equal form, i.e., `--path=-snakes-`, they just can't use the shortcut form.)

How do we pass string values containing spaces?  It needs to work on both Windows and Mac

**Comment by elliot-nelson** [comment]
> > (Users can still pass string values starting with dashes by using an equal form, i.e., `--path=-snakes-`, they just can't use the shortcut form.)
> 
> How do we pass string values containing spaces? It needs to work on both Windows and Mac

I don't think that will be a problem; basic double-quotes work the same (within reason) on both platforms.

```console
node myprogram.js -f one two "hello world" purple
```

```javascript
// On both Windows and Mac, this produces:
//   ["-f", "one", "two", "hello world", "purple"]
```

There's a lot of ways you could break Windows command line (by that I mean, lots of other things are different in the shell), but I think that's up to the Windows dev to be aware of.
